packages/react-google-maps-api/src/utils/useIsMounted.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react'
+
+const useIsMounted = () => {
+  const isMountedRef = React.useRef(false)
+  React.useEffect(function trackMountedState() {
+    isMountedRef.current = true
+    return () => {
+      isMountedRef.current = false
+    }
+  }, [])
+  return () => isMountedRef.current
```

Comment:
Please, let's drop a function completely. There is no benefit of hiding it in there and on contrary, you need to keep in deps of callbacks & effects. When you return the whole ref, it will always have the latest value

Reply:
ok, but that was actually the point of useCallback in the first place to return a constant value over time.

But if you prefer a ref i really dont care much

Reply:
Point is that even with useCallback you need to have in deps just in case it's recreated for some reason. Refs are different, they have a stable reference by default.